packages/integration-tests/src/EntityManager.test.ts

```diff
@@ -1023,8 +1023,105 @@ describe("EntityManager", () => {
+    const em = newEntityManager();
+
+    // Given an entity with a reference to another entity
+    const a1 = new Author(em, { firstName: "a1" });
+    const b1 = new Book(em, { title: "b1", author: a1 });
+    await em.flush();
+
+    // When we clone that entity and its reference
+    const a2 = await em.clone(a1, "books");
+    await em.flush();
+
+    // Then we expect the cloned entity to have a cloned copy of the original's reference
+    expect(a2.books.get[0].title).toEqual(b1.title);
+    expect(a2.books.get[0].id).not.toEqual(b1.id);
+  });
+
+  it("cannot clone many-to-many references", async () => {
+    const em = newEntityManager();
+
+    // Given an entity with a reference to another entity with a many-to-many reference
+    const a1 = new Author(em, { firstName: "a1" });
+    const b1 = new Book(em, { title: "b1", author: a1 });
+    const t1 = new Tag(em, { name: "t1", books: [b1] });
+    await em.flush();
+
+    // When we clone that entity and its nested references, which include a many-to-many reference
+    const promise = async function () {
+      await em.clone(a1, { books: "tags" });
+    };
+
+    // Then we expect the cloning to fail
+    await expect(promise).rejects.toThrow("Irreversible relation: tags");
+  });
+
+  it("can clone nested references", async () => {
+    const em = newEntityManager();
+
+    // Given an entity with a reference to another entity with a one-to-one
+    const a1 = new Author(em, { firstName: "a1" });
+    const b1 = new Book(em, { title: "b1", author: a1 });
+    const i1 = new Image(em, { fileName: "11", type: ImageType.BookImage, book: b1 });
+    await em.flush();
+
+    // When we clone that entity and its nested references
+    const a2 = await em.clone(a1, { books: "image" });
+    await em.flush();
+
+    // Then we expect the cloned entity to have cloned copies of all its nested references
+    const b2 = (await a2.books.load())[0];
+    const i2 = await b2.image.load();
+    expect(i2).toBeTruthy();
+    expect(i2?.id).not.toEqual(i1.id);
+    expect(i2?.fileName).toEqual(i1.fileName);
+    expect(i2?.type).toEqual(i1.type);
+    expect(await numberOf(em, Author, Book, Image)).toEqual([2, 2, 2]);
+  });
+
+  it("should only clone referenced entities when specified", async () => {
+    const em = newEntityManager();
+
+    // Given an entity with a reference to another entity
+    const a1 = new Author(em, { firstName: "a1", books: [newBook(em)] });
+    await em.flush();
+
+    // When we clone that entity and don't pass a populate hint for the reference
+    const a2 = await em.clone(a1);
+    await em.flush();
+
+    // Then we expect the cloned entity to have no references
+    expect(await a2.books.load()).toHaveLength(0);
+  });
 });
 
 function delay(ms: number): Promise<void> {
   return new Promise((resolve) => setTimeout(resolve, ms));
 }
+
+async function numberOf(em: EntityManager, ...args: EntityConstructor<any>[]): Promise<number[]> {
+  return await Promise.all(
```

Comment:
Really minor, you can merge w/o changing, but when you are returning a promise, you don't need to await it, i.e.:

```
return Promise.all(...)
```

Would be fine

Reply:
Ohhh that makes sense. I was worried that I'd end up with a nested promise being returned by `numberOf`